### PR TITLE
validate graph is not empty

### LINKED ISSUE #116: Empty Graph Validation
Orchestrator should prevent empty graphs from stitching and throw an exception 

**To Reproduce**

**Expected behavior**

**Additional context**

### PR DESCRIPTION
**Pull request by kmoore-intuit**
# What Changed

# Why
Closes #116 
Todo:

- [ ] Add tests
- [ ] Add docs

### CODE REVIEW

File: src/main/java/com/intuit/graphql/orchestrator/stitching/XtextStitcher.java [hunk 1]
```diff
@@ -161,6 +164,21 @@ public RuntimeGraph stitch(List<ServiceProvider> serviceProviders) {
...
+            .stream()
+            .map(ObjectTypeDefinition::getFieldDefinition)
+            .allMatch(List::isEmpty);
+
+    if(emptyGraph &&
+        !(xtextGraph.getServiceProvider().isFederationProvider() &&
+                getAllTypes(xtextGraph.getXtextResourceSet()).findAny().isPresent()
+        )
+    ) {
+      throw new StitchingException("Graph cannot be empty");
```
Comment (ashpak-shaikh) [inline src/main/java/com/intuit/graphql/orchestrator/stitching/XtextStitcher.java:178]: Maybe more context in here like the namespace

File: src/main/java/com/intuit/graphql/orchestrator/stitching/XtextStitcher.java [hunk 2]
```diff
@@ -106,6 +108,7 @@ public RuntimeGraph stitch(List<ServiceProvider> serviceProviders) {
     Map<String, XtextGraph> xtextGraphMap = serviceProviders.stream()
         .map(XtextGraphBuilder::build)
         .map(xtextGraph -> transform(xtextGraph, preMergeTransformers))
+        .peek(this::validateGraph)
```
Comment (CNAChino) [inline src/main/java/com/intuit/graphql/orchestrator/stitching/XtextStitcher.java:111]: just curious why validateGraph is after transform?

File: src/test/groovy/com/intuit/graphql/orchestrator/integration/TopLevelStitchingSpec.groovy [hunk 3]
```diff
@@ -126,4 +129,91 @@ class TopLevelStitchingSpec extends BaseIntegrationTestSpecification {
...
+        createGraphQLOrchestrator(emptyProvider)
+
+        then:
+        thrown(StitchingException.class)
+    }
+
+    def "Provider with empty operation and extending operation can stitch"() {
+        given:
+
+        String schema = "type Query {} extend type Query {getID: ID}"
```
Comment (CNAChino) [inline src/test/groovy/com/intuit/graphql/orchestrator/integration/TopLevelStitchingSpec.groovy:166]: not sure if needed, should we add test for empty operation.. e.g. `type Query {}`, i.e. no `extend type Query`

File: src/main/java/com/intuit/graphql/orchestrator/stitching/XtextStitcher.java (hunk 2 above)
Comment (kmoore-intuit) [reply src/main/java/com/intuit/graphql/orchestrator/stitching/XtextStitcher.java:111]: Needs to be run after the type extension transformer. 
For example, if someone's schema looks 
type Query{}
extend type Query { getA: ID }
this is consider empty when first getting the resources and looking at the query, mutation, and subscriptions

File: src/test/groovy/com/intuit/graphql/orchestrator/integration/TopLevelStitchingSpec.groovy (hunk 3 above)
Comment (kmoore-intuit) [reply src/test/groovy/com/intuit/graphql/orchestrator/integration/TopLevelStitchingSpec.groovy:182]: added a test case for that

**Review by ashpak-shaikh — APPROVED** [review]
